cmd/daemon/api_server.go

```diff
@@ -420,20 +434,17 @@ func (s *ApiServer) serve() {
 		if r.Method == "GET" {
 			s.handleRequest(ApiRequest{Type: ApiRequestTypeGetVolume}, w)
 		} else if r.Method == "POST" {
-			var data struct {
-				Volume uint32 `json:"volume"`
-			}
+			var data ApiRequestDataVolume
 			if err := json.NewDecoder(r.Body).Decode(&data); err != nil {
 				w.WriteHeader(http.StatusBadRequest)
 				return
 			}
-
-			if data.Volume < 0 {
+			if !data.Relative && data.Volume < 0 {
```

Comment:
This check wasn't actually working before: it was checking whether an unsigned integer was negative (it never is). With the change, it does in fact check for negative volume values.

Reply:
Ooops :)